system-addon/lib/TelemetryFeed.jsm

```diff
@@ -24,23 +27,47 @@ this.TelemetryFeed = class TelemetryFeed {
   }
 
   async init() {
+    Services.obs.addObserver(this.browserOpenNewtabStart, "browser-open-newtab-start");
+
     // TelemetrySender adds pref observers, so we initialize it after INIT
     this.telemetrySender = new TelemetrySender();
 
     const id = await ClientID.getClientID();
     this.telemetryClientId = id;
   }
 
+  browserOpenNewtabStart() {
+    perfService.mark("browser-open-newtab-start");
+  }
+
   /**
    * addSession - Start tracking a new session
    *
    * @param  {string} id the portID of the open session
+   * @param  {number} absVisChangeTime absolute timestamp of
+   *                                   document.visibilityState becoming visible
    */
-  addSession(id) {
+  addSession(id, absVisChangeTime) {
+    // XXX note that there is a race condition here; we're assuming that the
+    // most recent tab
+    let absBrowserOpenTabStart =
+      perfService.getMostRecentAbsMarkStartByName("browser-open-newtab-start");
```

Comment:
It's already a public API point, I'm not convinced it really buys us much.  Search-and-replace for something like this is awfully easy within a file, after all.